api/models/Centres.js

```diff
@@ -56,61 +56,63 @@ const model = {
-    },
-    female_active_movements_out: {
-      collection: 'movement',
-      note: 'this is a workaround until waterline supports conditional joins see balderdashy/waterline#988 and balderdashy/waterline#645',
-      via: 'active_female_centre_out'
-    },
     mo_type: {
       type: 'string',
       required: true,
       defaultsTo: 0
     },
+    events: {
+      collection: 'event',
+      via: 'centre'
+    },
+    detainees: {
+      collection: 'detainee',
+      via: 'centre'
+    },
+    movements: {
+      collection: 'movement',
+      via: 'centre'
+    },
     toJSON: function () {
-      const maleCapacity = this.male_capacity - this.male_in_use;
-      const femaleCapacity = this.female_capacity - this.female_in_use;
+      const unreconciledEventCounter = (gender, operations) => this.unreconciledEvents.reduce((count, event) => {
+        if (event.detainee.gender === gender && operations.indexOf(event.operation) > 0-1) {
+          return count + 1;
+        }
+        return count;
+      }, 0);
+      const unreconciledMovementCounter = (gender, direction) => this.unreconciledMovements.reduce((count, movement) => {
+        if (movement.gender === gender && movement.direction === direction) {
+          return count + 1;
+        }
+        return count;
+      }, 0);
+
       const response = {
         type: 'centre',
         id: this.id.toString(),
         attributes: {
           cidReceivedDate: this.cid_received_date,
           updated: this.updatedAt,
           heartbeatRecieved: this.heartbeat_recieved ? this.heartbeat_recieved.toString() : null,
-          name: this.name,
-          maleCapacity: this.male_capacity,
-          femaleCapacity: this.female_capacity,
-          maleInUse: this.male_in_use,
-          femaleInUse: this.female_in_use,
-          maleOutOfCommission: this.male_out_of_commission,
-          femaleOutOfCommission: this.female_out_of_commission,
-          maleAvailability: maleCapacity - this.male_out_of_commission,
-          femaleAvailability: femaleCapacity - this.female_out_of_commission,
-          maleActiveMovementsIn: this.male_active_movements_in.length,
-          maleActiveMovementsOut: this.male_active_movements_out.length,
-          femaleActiveMovementsIn: this.female_active_movements_in.length,
-          femaleActiveMovementsOut: this.female_active_movements_out.length
+          name: this.name
         },
         links: this.modelLinks('centres', reverseRouteService)
       };
+      ['male', 'female'].forEach((gender) => {
+        response.attributes[gender + 'Capacity'] = this[gender + '_capacity'];
+        response.attributes[gender + 'InUse'] = this[gender + '_in_use'];
+        response.attributes[gender + 'OutOfCommission'] = this[gender + '_out_of_commission'];
+        response.attributes[gender + 'Availability'] = this[gender + '_capacity'] - this[gender + '_in_use'] - this[gender + '_out_of_commission'];
+        if (this.reconciled) {
```

Comment:
Well, I didn't what to reconcile every time we toJSON.

The reconciliation data is transient. It depends on what scope/date range you want to perform the reconciliation for.

BedCountService has a performConfiguredReconciliation.

Reconciliation is one of the more computationally expensive operations we do, so I chose to be selective about how it's applied. If you'd like it done every time we toJSON, it can be done.